test/SgidClient.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { readFileSync } from 'fs'
+
+import SgidClient from '../src'
+
+import {
+  MOCK_ACCESS_TOKEN,
+  MOCK_API_VERSION,
+  MOCK_AUTH_CODE,
+  MOCK_AUTH_ENDPOINT,
+  MOCK_CLIENT_ID,
+  MOCK_CLIENT_PRIVATE_KEY,
+  MOCK_CLIENT_SECRET,
+  MOCK_HOSTNAME,
+  MOCK_REDIRECT_URI,
+  MOCK_SUB,
+  MOCK_USERINFO_PLAINTEXT,
+} from './mocks/constants'
+import {
+  tokenHandlerNoSub,
+  tokenHandlerNoToken,
+  userInfoHandlerMalformedData,
+  userInfoHandlerMalformedKey,
+  userInfoHandlerNoData,
+  userInfoHandlerNoKey,
+} from './mocks/handlers'
+import { server } from './mocks/server'
+
+const DEFAULT_SCOPE = 'myinfo.nric_number openid'
+const DEFAULT_RESPONSE_TYPE = 'code'
+
+describe('SgidClient', () => {
+  let client: SgidClient
+
+  beforeEach(() => {
```

Comment:
I'd use beforeAll instead of beforeEach since the client remains static throughout entire spec and the mock values are constants.

```suggestion
  beforeAll(() => {
```

Reply:
[8fa730f](https://github.com/opengovsg/sgid-client/pull/24/commits/8fa730fbe476cae80ad3c699c9c941b615a410b9)